pkg/server/api/status.go

```diff
@@ -16,17 +18,60 @@ func CreateStatus(code codes.Code, format string, a ...interface{}) *types.Statu
 	}
 }
 
-// StatusFromError creates a proto Status from given error
-func StatusFromError(err error) *types.Status {
-	if err == nil {
+// OK creates a success proto status
+func OK() *types.Status {
+	return CreateStatus(codes.OK, codes.OK.String())
+}
+
+// MakeStatus logs and returns a status composed of: msg, err and code.
+// Errors are treated differentially according to its gRPC code.
+func MakeStatus(log logrus.FieldLogger, code codes.Code, msg string, err error) *types.Status {
+	e := MakeErr(log, code, msg, err)
+	if e == nil {
+		return OK()
+	}
+
+	return CreateStatus(code, status.Convert(e).Message())
+}
+
+// MakeErr logs and returns an error comoposed of: msg, err and code.
+// Errors are treated differentially according to its gRPC code.
+func MakeErr(log logrus.FieldLogger, code codes.Code, msg string, err error) error {
+	errMsg := msg
+	switch code {
+	case codes.OK:
+		// Log 'OK' status at INFO level and return nil error
+		log.Info(capitalize(msg))
 		return nil
+
+	case codes.InvalidArgument:
+		// Add the prefix 'Invalid argument' for InvalidArgument errors
+		if err != nil {
+			log = log.WithError(err)
+			errMsg = fmt.Sprintf(msg+": %v", err)
+		}
+
+		log.Error("Invalid argument: " + msg)
+		return status.Error(code, errMsg)
+
+	case codes.NotFound:
+		// Do not log nor return the inner error for NotFound errors
+		log.Error(capitalize(msg))
+		return status.Error(code, errMsg)
```

Comment:
no changes on errMsg you can use msg here

Reply:
Right, both can be used. I'd rather use `errMsg` here for consistency: 
`msg`      ->  for all the logged messages
`errMsg` ->  for the returned error.